circle.yml

```diff
@@ -1,38 +1,109 @@
-machine:
-  services:
-    - docker
-  hosts:
-    olympia.dev: 127.0.0.1
+version: 2.0
 
-dependencies:
-  override:
-    - docker version
-    - >
-        printf '{"commit":"%s","version":"%s","source":"https://github.com/%s/%s","build":"%s"}\n'
-        "$CIRCLE_SHA1"
-        "$CIRCLE_TAG"
-        "$CIRCLE_PROJECT_USERNAME"
-        "$CIRCLE_PROJECT_REPONAME"
-        "$CIRCLE_BUILD_URL"
-        > version.json
-    - docker build -t app:build -f Dockerfile.deploy .
-
-test:
-  override:
-    - echo "UI Tests are run on Travis CI. https://travis-ci.org/mozilla/addons-server/"
-
-deployment:
-  latest:
-    branch: master
-    commands:
-      - docker tag app:build ${DOCKERHUB_REPO}:latest
-      - docker login -e $DOCKERHUB_EMAIL -u $DOCKERHUB_USER -p $DOCKERHUB_PASS
-      - docker push ${DOCKERHUB_REPO}:latest
-
-  releases:
-    # push all tags
-    tag: /.*/
-    commands:
-      - docker tag app:build ${DOCKERHUB_REPO}:${CIRCLE_TAG}
-      - docker login -e $DOCKERHUB_EMAIL -u $DOCKERHUB_USER -p $DOCKERHUB_PASS
-      - docker push ${DOCKERHUB_REPO}:${CIRCLE_TAG}
+jobs:
+  build:
+    machine: true
+    working_directory: ~/addons-server
+    steps:
+      - checkout
+      - run: >
+          printf '{"commit":"%s","version":"%s","source":"https://github.com/%s/%s","build":"%s"}\n'
+          "$CIRCLE_SHA1"
+          "$CIRCLE_TAG"
+          "$CIRCLE_PROJECT_USERNAME"
+          "$CIRCLE_PROJECT_REPONAME"
+          "$CIRCLE_BUILD_URL"
+          > version.json
+      - run: docker build -t app:build -f Dockerfile.deploy .
+      - persist_to_workspace:
+          root: .
+          paths: .
+  integration_test:
+    working_directory: ~/addons-server
+    machine: true
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - uitest-cache-{{ checksum "requirements/docs.txt" }}
+            - uitest-cache-{{ checksum "requirements/prod.txt" }}
+            - uitest-cache-{{ checksum "requirements/dev.txt" }}
+            - uitest-cache-{{ checksum "tests/ui/requirements.txt" }}
+            - uitest-cache-
```

Comment:
Guess that's one key too much?

Reply:
Heh, so this means that the cache will not be restored if any of those files change. 

See: https://circleci.com/docs/2.0/caching/

Reply:
But that last key does not reference any file 😕

Reply:
If the cache gets changed it updates the key, so when it restores the cache, even if one of the files are not changed the cache stored at key would have been changed. Confusing kinda but the docs explain it better.